packages/core-container/__tests__/registrars/plugin.test.js

```diff
@@ -73,7 +73,7 @@ describe('Plugin Registrar', () => {
   })
 
   describe('tearDown', () => {
-    xit('should deregister all the plugins in inverse order', async () => {
+    it('should deregister plugins supporting deregister', async () => {
```

Comment:
Everything looks OK, but I'll split this test into 2 different tests:
 1. "in inverse order": it's important that this is done in a predictable way
 2 "plugins supporting deregister"

Reply:
it's a little bit more difficult the inverse order. These "deregister" callbacks for each plugin are not linekd to each other, so from the outside perspective, they're all unregistered at once. You'd have to have the plugins know of a shared collection(an array or so) such that when 'deregister' is called, they push their plugin name into the collection.

Then you'd have to have a way to propagate that collection to the test so that it can create assertions on that.

It seems like a convoluted method just to test this, seemingly, atomic operation.

Reply:
It's simpler than that: all the registered plugins that are "deregisterable" belong to the `this.deregister` array.
Each time a new plugin is registered, they're pushed to that Array, so they keep the order.

The current test doesn't check that, but it could be renamed as the 2 option, as you had done, while adding a new one to deal with the first case.

Apart from that, a supplementary test would be testing that registering a new plugin, adds it to the `deregister` array.